api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/post-schema-create'
+        required: true
+  '/posts/{post_id}':
+    get:
+      tags:
+        - posts
+      parameters:
+        - description: The Invotra UUID of the post.
+          in: path
+          name: post_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: Returns a single complete post object.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/post-schema'
+        '400':
+          description: The supplied UUID was malformed.
+        '404':
+          description: The UUID does not correspond to an existing post.
+      description: Get information about the post.
+    put:
+      tags:
+        - posts
+      parameters:
+        - description: The Invotra UUID of the post.
+          in: path
+          name: post_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The post information was updated successfully.
+        '400':
+          description: The supplied UUID or JSON was malformed.
+        '404':
+          description: The UUID does not correspond to an existing post.
+        '422':
+          description: |
+            The post was not updated due to a Drupal validation failure or a
+            JSON schema validation failure.
+      description: Update post information.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/post-schema-base'
+        required: true
+  '/posts/{post_id}/likes':
+    get:
+      tags:
+        - posts
+      parameters:
+        - description: The Invotra UUID of the post.
+          in: path
+          name: post_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - $ref: '#/components/parameters/offsetParam'
+        - $ref: '#/components/parameters/limitParam'
+      responses:
+        '200':
+          description: |
+            List of likes for a post.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/user-schema'
+        '400':
+          description: The supplied UUID was malformed.
+        '404':
+          description: The UUID does not correspond to an existing post.
+      description: Get a list of likes for a post.
```

Comment:
I think we should clarify what a "like" actually is - in this case, a user (who liked the post)

I know this can be inferred from the schema but no harm to describe it too :)

Reply:
updated